sql/sql_update.cc

```diff
@@ -529,6 +529,7 @@ int mysql_update(THD *thd,
   if (!(select && select->quick))
     status_var_increment(thd->status_var.update_scan_count);
 
+  IO_CACHE tempfile;
```

Comment:
It may work, but we still have select->file members pointing to tempfile. This has to be reviewed carefully and if we decide it is safe, we'd better add a comment explaining why this hack is safe.